lab_6_pipeline/pipeline.py

```diff
@@ -62,11 +126,23 @@ def __init__(
             corpus_manager (CorpusManager): CorpusManager instance
             analyzer (LibraryWrapper | None): Analyzer instance
         """
+        self._corpus = corpus_manager
+        self.analyzer = analyzer
 
     def run(self) -> None:
         """
         Perform basic preprocessing and write processed text to files.
         """
+        documents = []
+        if self.analyzer:
+            documents = self.analyzer.analyze([article.text for article
+                                               in self._corpus.get_articles().values()])
+
+        for num, article in enumerate(self._corpus.get_articles().values()):
+            to_cleaned(article)
+            if self.analyzer and documents:
+                article.set_conllu_info(documents[num])
+                self.analyzer.to_conllu(article)
```

Comment:
self._corpus.get_values... is called twice in this code

can you rewrite code to call get_articles() just once?

Reply:
i added the value articles where i save the result from executing self._corpus.get_articles()